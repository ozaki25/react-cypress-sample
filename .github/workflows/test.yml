name: Test
on: [pull_request]
jobs:
  unit-test:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v1
      - name: Use Node.js 12
        uses: actions/setup-node@v1
        with:
          node-version: 12
      - name: install
        run: |
          yarn install
      - name: lint
        run: |
          yarn lint
      - name: test
        run: |
          yarn test
  e2e-test:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v1
      - name: Use Node.js 12
        uses: actions/setup-node@v1
        with:
          node-version: 12
      - name: install
        run: |
          yarn install
      - name: e2e-test
        run: |
          yarn e2e
