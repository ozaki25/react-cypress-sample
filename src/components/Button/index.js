import React from 'react';
import styled from 'styled-components';

const Btn = styled.button`
  font-size: 1em;
  font-weight: 600;
  color: #1565c0;
  padding: 0.3em 0.8em;
  background: #fafafa;
  border: 1px solid #1565c0;
  border-radius: 0.1em;

  &:hover,
  &:active,
  &:focus {
    color: #fafafa;
    background-color: #1565c0;
  }
`;

function Button(props) {
  return <Btn {...props} />;
}

export default Button;
