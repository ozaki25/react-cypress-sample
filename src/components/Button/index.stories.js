import React from 'react';
import { action } from '@storybook/addon-actions';
import Button from '.';

export default {
  title: 'Button',
};

export const nomal = () => (
  <Button onClick={action('clicked')}>Hello Button</Button>
);
