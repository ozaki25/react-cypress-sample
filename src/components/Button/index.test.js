import React from 'react';
import { snapshot } from '../../test/helper';
import Button from '.';

snapshot('Button', <Button>ボタン</Button>);
