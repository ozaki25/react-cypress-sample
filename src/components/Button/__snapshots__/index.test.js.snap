// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot: Button 1`] = `
.c0 {
  font-size: 1em;
  font-weight: 600;
  color: #1565c0;
  padding: 0.3em 0.8em;
  background: #fafafa;
  border: 1px solid #1565c0;
  border-radius: 0.1em;
}

.c0:hover,
.c0:active,
.c0:focus {
  color: #fafafa;
  background-color: #1565c0;
}

<button
  className="c0"
>
  ボタン
</button>
`;
